sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -515,13 +515,33 @@ class Analyzer(
                 s"Aggregate expression required for pivot, found '$e'")
           }
         }
+        // Check all pivot values are literal and match pivot column data type.
+        val evalPivotValues = pivotValues.map { value =>
+          if (!Cast.canCast(value.dataType, pivotColumn.dataType)) {
+            throw new AnalysisException(s"Invalid pivot value '$value': " +
+              s"value data type ${value.dataType.simpleString} does not match " +
+              s"pivot column data type ${pivotColumn.dataType.simpleString}")
+          }
+          try {
+            Cast(value, pivotColumn.dataType).eval(EmptyRow)
+          } catch {
+            case _: UnsupportedOperationException =>
+              throw new AnalysisException(
+                s"Literal expressions required for pivot values, found '$value'")
```

Comment:
Is `UnsupportedOperationException` raised only in the case if `value` is not a literal. Probably you can check that it is a literal earlier?

Reply:
Yes, you are right. Please refer to @gatorsmile's comment.